File: 119-pascals-triangle-ii/119-pascals-triangle-ii.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        prev, cur = [], []
        for row in range(rowIndex+1):
            cur = []
            for i in range(1+row):
                if i == 0 or i == row:
                    cur.append(1)
                else:
                    cur.append(prev[i-1]+prev[i])
            prev = cur[:]
        return cur
        

File: 119-pascals-triangle-ii/119-pascals-triangle-ii.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> prev = new LinkedList<>(Arrays.asList(1));
        if(rowIndex == 0)return prev;
        for(int i = 1; i <= rowIndex; i++){
            List<Integer> temp = new LinkedList<>();
            temp.add(1);
            for(int j = 0; j < prev.size() - 1; j++){
                temp.add(prev.get(j) + prev.get(j+1));
            }
            temp.add(1);
            prev = temp;
        }
        return prev;
        
        
    }
}